src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -639,6 +639,119 @@ void SpatialPooler::stripUnlearnedColumns(UInt activeArray[]) const
   }
 }
 
+bool SpatialPooler::operator==(const SpatialPooler &sp) const
+{
+    const UInt numColumns = this->getNumColumns();
+    const UInt numInputs = this->getNumInputs();
+
+    ASSERT_TRUE(this->getNumColumns() == sp.getNumColumns());
+    ASSERT_TRUE(this->getNumInputs() == sp.getNumInputs());
+    ASSERT_TRUE(this->getPotentialRadius() ==
+              sp.getPotentialRadius());
+    ASSERT_TRUE(this->getPotentialPct() == sp.getPotentialPct());
+    ASSERT_TRUE(this->getGlobalInhibition() ==
+              sp.getGlobalInhibition());
+    ASSERT_TRUE(this->getNumActiveColumnsPerInhArea() ==
+              sp.getNumActiveColumnsPerInhArea());
+    ASSERT_TRUE(almost_eq(this->getLocalAreaDensity(),
+              sp.getLocalAreaDensity()));
+    ASSERT_TRUE(this->getStimulusThreshold() ==
+              sp.getStimulusThreshold());
+    ASSERT_TRUE(this->getDutyCyclePeriod() == sp.getDutyCyclePeriod());
+    ASSERT_TRUE(almost_eq(this->getBoostStrength(), sp.getBoostStrength()));
+    ASSERT_TRUE(this->getIterationNum() == sp.getIterationNum());
+    ASSERT_TRUE(this->getIterationLearnNum() ==
+              sp.getIterationLearnNum());
+    ASSERT_TRUE(this->getSpVerbosity() == sp.getSpVerbosity());
+    ASSERT_TRUE(this->getWrapAround() == sp.getWrapAround());
+    ASSERT_TRUE(this->getUpdatePeriod() == sp.getUpdatePeriod());
+    ASSERT_TRUE(almost_eq(this->getSynPermTrimThreshold(),
+              sp.getSynPermTrimThreshold()));
+    cout << "check: " << this->getSynPermActiveInc() << " " <<
+      sp.getSynPermActiveInc() << endl;
+    ASSERT_TRUE(almost_eq(this->getSynPermActiveInc(),
+              sp.getSynPermActiveInc()));
+    ASSERT_TRUE(almost_eq(this->getSynPermInactiveDec(),
+              sp.getSynPermInactiveDec()));
+    ASSERT_TRUE(almost_eq(this->getSynPermBelowStimulusInc(),
+              sp.getSynPermBelowStimulusInc()));
+    ASSERT_TRUE(almost_eq(this->getSynPermConnected(),
+              sp.getSynPermConnected()));
+    ASSERT_TRUE(almost_eq(this->getMinPctOverlapDutyCycles(),
+              sp.getMinPctOverlapDutyCycles()));
+
+
+    auto boostFactors1 = new Real[numColumns];
+    auto boostFactors2 = new Real[numColumns];
+    this->getBoostFactors(boostFactors1);
+    sp.getBoostFactors(boostFactors2);
+    ASSERT_TRUE(check_vector_eq(boostFactors1, boostFactors2, numColumns));
+    delete[] boostFactors1;
+    delete[] boostFactors2;
```

Comment:
This seems like some optimization. Q: keep the delete[] (saves memory), or remove it (saves cycles, as the local memory is unallocated anyway at the end of the function/scope)